db/db_impl/db_impl.cc

```diff
@@ -5536,6 +5591,36 @@ Status DBImpl::GetCreationTimeOfOldestFile(uint64_t* creation_time) {
     return Status::NotSupported("This API only works if max_open_files = -1");
   }
 }
+
+void DBImpl::RecordSeqnoToTimeMapping() {
+  SequenceNumber min_first_mem_seqno = kMaxSequenceNumber;
+
+  for (auto cfd : *versions_->GetColumnFamilySet()) {
+    if (cfd->ioptions()->preclude_last_level_data_seconds > 0) {
+      min_first_mem_seqno =
+          std::min(cfd->GetFirstMemtableSequenceNumber(), min_first_mem_seqno);
+    }
+  }
```

Comment:
Is min_first_mem_seqno only for binary search hint? If that is the case, it feels too complex and might not necessarily to be cheaper. Some applications might have thousands of CFs. Looping through all CFs is generally encouraged to be avoided if possible, especially within DB mutex (I know it is now outside but still).

Reply:
That makes sense. I'm going to remove that part for now as it's just an optimization to reduce the memory usage. As now the memory usage is capped at 1.6K per DB instance `(100 * (8+8))` or 16K for the worst case (multiple CFs with different settings). It's not worth doing I think.

Reply:
Oh I got it. It is for retention. I think there is another counter for that. In that case, this number is better to be updated in flush/compaction path where we already loop through all CFs for something else, just like VersionSet::PreComputeMinLogNumberWithUnflushedData(). The information can be used by something else. We might even have the information precalculated, though I didn't find it.

Of course we don't have to do it here.